include/sdbus-c++/Types.h

```diff
@@ -237,6 +239,9 @@ namespace sdbus {
 
         UnixFd& operator=(UnixFd&& other)
         {
+            if (this == &other) {
+                return *this;
+            }
             close();
             fd_ = other.fd_;
             other.fd_ = -1;
```

Comment:
```suggestion
            fd_ = std::exchange(other.fd_, -1);
```
Can also use this in `release()`.